### PR TITLE
Ceres-Solver: new recipe

### PR DESCRIPTION
**Pull request by extrowerk**

### CODE REVIEW

**Comment by extrowerk** [comment]
https://github.com/ceres-solver/ceres-solver/issues/649

File: sci-libs/ceres_solver/ceres_solver-2.0.0.recipe [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+defineDebugInfoPackage ceres_solver$secondaryArchSuffix \
+	$libDir/libceres.so.$portVersion
+
+BUILD()
+{
+	mkdir -p build && cd "$_"
+
+	cmake .. \
+		-DCMAKE_INSTALL_PREFIX=$prefix \
```
Comment (korli) [inline sci-libs/ceres_solver/ceres_solver-2.0.0.recipe:81]: isn't CMAKE_INSTALL_PREFIX part of cmakeDirArgs?

Comment (Begasus) [reply sci-libs/ceres_solver/ceres_solver-2.0.0.recipe:81]: IIRC they don't use cmakeDirArgs (GNUInstallDirs)

Comment (korli) [reply sci-libs/ceres_solver/ceres_solver-2.0.0.recipe:81]: > IIRC they don't use cmakeDirArgs (GNUInstallDirs)

This recipe uses cmakeDirArgs

Comment (Begasus) [inline sci-libs/ceres_solver/ceres_solver-2.0.0.recipe:81]: This should fix it:
```diff
+	-DCMAKE_INSTALL_INCLUDEDIR=$includeDir \
+	-DCMAKE_INSTALL_LIBDIR=$libDir \
```

File: sci-libs/ceres_solver/ceres_solver-2.0.0.recipe [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		-DBUILD_SHARED_LIBS=ON \
+		-DEigen3_DIR=`finddir B_SYSTEM_DATA_DIRECTORY`/eigen3/cmake \
+		$cmakeDirArgs
```
Comment (Begasus) [inline sci-libs/ceres_solver/ceres_solver-2.0.0.recipe:84]:
```diff
-		-DEigen3_DIR=`finddir B_SYSTEM_DATA_DIRECTORY`/eigen3/cmake \
-		$cmakeDirArgs
+		-DEigen3_DIR=`finddir B_SYSTEM_DATA_DIRECTORY`/eigen3/cmake
```

File: sci-libs/ceres_solver/ceres_solver-2.0.0.recipe [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+INSTALL()
+{
+	cd build
+
+	make install $jobArgs
+
+	mkdir -p $includeDir
+	mv $prefix/include/* $includeDir/
+	rmdir $prefix/include
+
```
Comment (Begasus) [inline sci-libs/ceres_solver/ceres_solver-2.0.0.recipe:98]:
```diff
-	mkdir -p $includeDir
-	mv $prefix/include/* $includeDir/
-	rmdir $prefix/include
-
```

**Comment by Begasus** [comment]
Forgot I patched the source for this :)
```
From d6c1f99f3f4a2b38f349cf4aefbceefafbfce296 Mon Sep 17 00:00:00 2001
From: begasus <begasus@gmail.com>
Date: Wed, 30 Dec 2020 18:19:10 +0000
Subject: Set paths for install


diff --git a/CMakeLists.txt b/CMakeLists.txt
index ea7e9b8..a415b8f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -661,21 +661,21 @@ endif (BUILD_EXAMPLES)
 
 # Setup installation of Ceres public headers.
 file(GLOB CERES_HDRS ${Ceres_SOURCE_DIR}/include/ceres/*.h)
-install(FILES ${CERES_HDRS} DESTINATION include/ceres)
+install(FILES ${CERES_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/ceres)
 
 file(GLOB CERES_PUBLIC_INTERNAL_HDRS ${Ceres_SOURCE_DIR}/include/ceres/internal/*.h)
-install(FILES ${CERES_PUBLIC_INTERNAL_HDRS} DESTINATION include/ceres/internal)
+install(FILES ${CERES_PUBLIC_INTERNAL_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/ceres/internal)
 
 # Also setup installation of Ceres config.h configured with the current
 # build options into the installed headers directory.
 install(FILES ${Ceres_BINARY_DIR}/config/ceres/internal/config.h
-        DESTINATION include/ceres/internal)
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/ceres/internal)
 
 if (MINIGLOG)
   # Install miniglog header if being used as logging #includes appear in
   # installed public Ceres headers.
   install(FILES ${Ceres_SOURCE_DIR}/internal/ceres/miniglog/glog/logging.h
-          DESTINATION include/ceres/internal/miniglog/glog)
+          DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/ceres/internal/miniglog/glog)
 endif (MINIGLOG)
 
 # Ceres supports two mechanisms by which it can be detected & imported into
@@ -720,7 +720,7 @@ write_basic_package_version_file("${Ceres_BINARY_DIR}/CeresConfigVersion.cmake"
 if (WIN32)
   set(RELATIVE_CMAKECONFIG_INSTALL_DIR CMake)
 else ()
-  set(RELATIVE_CMAKECONFIG_INSTALL_DIR lib${LIB_SUFFIX}/cmake/Ceres)
+  set(RELATIVE_CMAKECONFIG_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}${LIB_SUFFIX}/cmake/Ceres)
 endif ()
 
 # This "exports" for installation all targets which have been put into the
-- 
2.28.0
```

**Comment by Begasus** [comment]
@extrowerk anything new here (so far no changes to the comments)?

**Comment by extrowerk** [comment]
I probably will have time this week to recheck

**Comment by extrowerk** [comment]
Now it seems to be ok, upstream switched to gnuinstalldirs.

**Pull request merged**